src/components/ProjectToolbar/index.jsx

```diff
@@ -0,0 +1,93 @@
+import cls from 'classnames';
+import { useEffect, useState } from 'react';
+import ProjectStatusIcon from '../ProjectStatusIcon';
+import ProjectStatusDek from '../ProjectStatusDek';
+import {
+  flex, layout, margin, colors, typography as type,
+} from '../../theme';
+import ButtonsGroup from './ButtonsGroup';
+import store from '../../store';
+
+export default function ProjectToolbar({ selectedProject, isArchive }) {
+  const [projectDetail, setProjectDetail] = useState({});
+  const [status, setStatus] = useState(undefined);
+  const [timestamp, setTimestamp] = useState(undefined);
+
+  // Gets project details when selected project changes
+  useEffect(() => {
+    (async () => {
+      if (!isArchive) {
+        const project = await store.getProject(selectedProject);
+        setProjectDetail(project);
+      } else {
+        setProjectDetail(selectedProject);
+      }
+    })();
+  }, [selectedProject]);
+
+  // Listener for changes to a project in the store (like backup time)
+  useEffect(() => {
```

Comment:
This effect is used in a couple different places, though not in the same exact way other than for listening for a change for a given project slug. But I'm thinking it may be useful as it's own hook?

Reply:
I think in general state management needs a second look at for this. I think honestly, it might be worth setting up atoms that handle all the connection to the store. The fact that we keep having to do async calls to get details and set up listeneres to update things as the store changes, makes me thing that's likely the way to go here. 

Taking a first pass something like
- Atom for projects list
- Atom for archive projects list
- Atom for active project
- Selector to consolidate differences in the archive and project lists
- Atom family for project details
- Atom family for archive project details. 

Let me know if you want to discuss this further, we can map it out properly.

Reply:
Maybe some stuff for the settings too.